observability/src/main/java/io/grpc/observability/interceptors/InternalLoggingChannelInterceptor.java

```diff
@@ -16,33 +16,248 @@
 
 package io.grpc.observability.interceptors;
 
+import com.google.protobuf.Duration;
+import com.google.protobuf.util.Durations;
 import io.grpc.CallOptions;
 import io.grpc.Channel;
 import io.grpc.ClientCall;
 import io.grpc.ClientInterceptor;
+import io.grpc.Context;
+import io.grpc.Deadline;
+import io.grpc.ForwardingClientCall.SimpleForwardingClientCall;
+import io.grpc.ForwardingClientCallListener.SimpleForwardingClientCallListener;
 import io.grpc.Internal;
+import io.grpc.Metadata;
 import io.grpc.MethodDescriptor;
+import io.grpc.Status;
+import io.grpc.internal.TimeProvider;
+import io.grpc.observability.logging.Sink;
+import io.grpc.observabilitylog.v1.GrpcLogRecord.EventLogger;
+import io.grpc.observabilitylog.v1.GrpcLogRecord.EventType;
+import java.util.UUID;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.logging.Level;
+import java.util.logging.Logger;
 
-/** A logging interceptor for {@code LoggingChannelProvider}. */
+/**
+ * A logging interceptor for {@code LoggingChannelProvider}.
+ */
 @Internal
 public final class InternalLoggingChannelInterceptor implements ClientInterceptor {
+  private static final Logger logger = Logger
+      .getLogger(InternalLoggingChannelInterceptor.class.getName());
+
+  private final LogHelper helper;
 
   public interface Factory {
     ClientInterceptor create();
   }
 
   public static class FactoryImpl implements Factory {
+    private final Sink sink;
+    private final LogHelper helper;
+
+    static LogHelper createLogHelper(Sink sink, TimeProvider provider) {
+      return new LogHelper(sink, provider);
+    }
+
+    public FactoryImpl(Sink sink) {
+      this.sink = sink;
+      this.helper = createLogHelper(sink, TimeProvider.SYSTEM_TIME_PROVIDER);
+    }
 
     @Override
     public ClientInterceptor create() {
-      return new InternalLoggingChannelInterceptor();
+      return new InternalLoggingChannelInterceptor(helper);
+    }
+
+    public void close() {
+      sink.close();
```

Comment:
check if sink is null? If sink is null just ignore (no need to throw or log error)

Reply:
Added null check